### PR TITLE
Support any

### LINKED ISSUE #18627: [jit] Python built-in function support
We support many built-in functions ([Python 2](https://docs.python.org/2/library/functions.html), [Python 3](https://docs.python.org/3/library/functions.html)), but not all of them. Some of these need some more discussion (i.e. what about builtins that exist in Python 2 but not 3 and vice versa), but many can be done piecemeal and should be straightforward to implement. When you implement a builtin please check to see if it is a [magic method](https://rszalski.github.io/magicmethods/).

To do
- [x] `abs()`
- [ ] `all()`
- [ ] `any()`
- [x] `bin()`
- [x] `bool()`
- [ ] `bytearray()`
- [x] `chr()`
- [x] `dict()`
- [x] `divmod()`
- [x] `enumerate()`
- [x] `float()`
- [x] `getattr()`
- [x] `hash()`
- [x] `hex()`
- [x] `int()`
- [x] `isinstance()`
- [x] `len()`
- [x] `list()`
- [x] `long()` (long is deprecated in py3 builtins)
- [x] `max()`
- [x] `min()`
- [x] `oct()`
- [x] `ord()`
- [x] `pow()`
- [x] `print()`
- [x] `range()`
- [x] `reversed()`
- [x] `round()`
- [ ] `set()`
- [ ] `slice()`
- [x] `sorted()` ([separate issue](https://github.com/pytorch/pytorch/issues/19372), implemented for lists)
- [x] `str()`
- [ ] `sum()`
- [ ] `type()`

Maybe to do later (when things like first class modules are in, or if we ever have something like iterators)
- [ ] `callable()`
- [x] `tuple()`
- [ ] `complex()`
- [ ] `delattr()`
- [ ] `filter()`
- [ ] `format()`
- [ ] `frozenset()`
- [ ] `globals()`
- [ ] `hasattr()`
- [ ] `issubclass()`
- [ ] `iter()`
- [ ] `locals()`
- [ ] `next()`
- [ ] `object()`
- [ ] `property()`
- [ ] `reduce()`
- [ ] `setattr()`
- [ ] `repr()`
- [ ] `super()`
- [ ] `vars()`
- [x] `zip()`

Probably won't do
- [ ] `__import__()`
- [ ] `basestring()`
- [ ] `classmethod()`
- [ ] `cmp()`
- [ ] `compile()`
- [ ] `dir()`
- [ ] `eval()`
- [ ] `execfile()`
- [ ] `file()`
- [ ] `help()`
- [ ] `id()`
- [ ] `input()`
- [ ] `map()`
- [ ] `memoryview()`
- [ ] `open()`
- [ ] `raw_input()`
- [ ] `reload()`
- [ ] `staticmethod()`
- [ ] `unichr()`
- [ ] `unicode()`
- [ ] `xrange()`

cc @suo

### PR DESCRIPTION
**Pull request by nikithamalgifb**
Fixes #18627

Summary:
Adds torch.any support for JIT

Test:
python test/test_jit.py -k test_torch_any
python test/test_jit.py -k test_any

### CODE REVIEW

File: test/test_jit.py [hunk 1]
```diff
@@ -9090,6 +9090,39 @@ def forward(self, x):
...
 
+    def test_torch_any(self):
+        def fn(x):
+            return torch.any(x)
+
+        self.checkScript(fn, (torch.randn(3, 4), ))
+        self.checkScript(fn, (torch.empty(3), ))
+        self.checkScript(fn, (torch.empty(1), ))
+        self.checkScript(fn, (torch.ones(3, 4),))
+        self.checkScript(fn, (torch.zeros(5, 7, 1),))
```
Comment (tugsbayasgalan) [inline test/test_jit.py:9101]: It looks like torch.any also supports slicing through specified dimension. (torch.any(a, 1)). Could we add a test case for that?

**Review by tugsbayasgalan — COMMENTED** [review]
Are we supporting a torch.any with dimension? Pytorch doc says torch.any supports dimensional slicing as well.

File: test/test_jit.py (hunk 1 above)
Comment (nikithamalgifb) [reply test/test_jit.py:9104]: Yea, agreed. fixed.

Comment (tugsbayasgalan) [reply test/test_jit.py:9104]: cool

**Review by tugsbayasgalan — APPROVED** [review]

**Pull request closed without merging**